File: search_file.dart
import 'dart:io';

void main() {
  final dir = Directory(r'c:\dukaanZone\flutter_app\lib');
  if (!dir.existsSync()) {
    print('Directory does not exist');
    return;
  }
  dir.listSync(recursive: true).forEach((entity) {
    if (entity is File && entity.path.endsWith('.dart')) {
      try {
        final lines = entity.readAsLinesSync();
        for (var i = 0; i < lines.length; i++) {
          final line = lines[i];
          if (line.contains('class ') && line.contains('Chat')) {
            print('${entity.path}:${i + 1}: ${line.trim()}');
          }
        }
      } catch (e) {
        // ignore
      }
    }
  });
}

File: search_file.py
import os

directory = r"c:\dukaanZone\flutter_app\lib"
for root, dirs, files in os.walk(directory):
    for file in files:
        if file.endswith('.dart'):
            filepath = os.path.join(root, file)
            try:
                with open(filepath, 'r', encoding='utf-8') as f:
                    for line_num, line in enumerate(f, 1):
                        if "class " in line and "Chat" in line:
                            print(f"{filepath}:{line_num}: {line.strip()}")
            except Exception as e:
                pass
